:lipstick: Make access easier to the companies' link

ykh opened the pull request:
- Move the Jobs column left next to the Name column
- Convert companies' name from plain text to hyperlink

Changes preview:
![image](https://github.com/SiaExplains/visa-sponsorship-companies/assets/2290912/9ea0c593-7a0e-4279-a087-710e30c8acc6)

ykh committed:
:lipstick: Make access easier to the companies' link

- Move the Jobs column left next to the Name column
- Convert companies' name from plain text to hyperlink

ykh commented:
@SiaExplains Just for remind.

ykh commented:
Anybody there!

SiaExplains commented on src/app/page.tsx:
```diff
@@ -106,6 +106,7 @@ export default function Home() {
         <table>
           <tbody>
             <tr>
+              <th>Jobs</th>
```
What if we change this `Jobs` heading? Since this app is not referring jobs and only it is showing companies that offered job offer to abroad employees, maybe we can remove this heading or rename it somethings more relevant. What do you think?

ykh replied:
Indeed, I think it's sort of redundant (not relevant).
So, I will send another tiny PR.

SiaExplains commented:
Thanks so much @ykh , Approved + one small suggestion.

ykh committed:
:lipstick: Remove non-relevant 'Jobs' head

- Move LinkedIn icon next to company name.
- Make whole space of Name cell clickable (easier to
 click on the link).

ykh commented:
Little new changes:
![image](https://github.com/SiaExplains/visa-sponsorship-companies/assets/2290912/318e2921-9db2-45b6-be9b-fe301245ae21)

Just for keep the previous user-experience I didn't remove the LinkedIn icon.

@SiaExplains
